lubot.js

```diff
@@ -175,23 +175,40 @@ bot.helpers.utils = {
     return false;
   },
   stripUpKarma: function(text) {
-    var re = new RegExp("([A-Za-z0-9]{1,})(?=[ >:]*\\+\\+)");
-    var matches = re.exec(text);
-    if (matches) {
-      return(matches[1]);
-    }
-    return false;
+    return bot.helpers.utils.stripKarma(text, 'up');
   },
   stripDownKarma: function(text) {
-    var re = new RegExp("([A-Za-z0-9]{1,})(?=[ >:]*\-\-)");
-    var matches = re.exec(text);
+    return bot.helpers.utils.stripKarma(text, 'down');
+  },
+  stripKarma: function(text, direction) {
+    var re = new RegExp('^(.+)(?=[ >:]*' + (direction === 'down' ? '--' : '\\+\\+') + '$)', 'i'),
```

Comment:
Should we strip the @ symbol if it is present?

```
'^@?(.+)...
```

Why is the end of the string inside the last capture group?

It'd be helpful to document the regex. This probably isn't accurate, but something like this:

```
// Starting at the beginning of the string, search for any set of characters
// ending in -- or ++, optionally with a trailing spaces, colons or carets.
```
Instead of concatenating strings, you can use util.format to do sprintf style replacements. If you think that's easier to read, that is… (see below)

If I'm reading this correctly, this wouldn't strip extra `+` or `-` from the end of the string, which means that q0rban--- would result in q0rban- getting karma reduced.  Is that correct? If so, what about this instead:

```js
var f = require('util').format;
var modifier = (direction === 'down') ? '-' : '\\+';
var re = new RegExp(f('^(.+)(?=[ >:]*%s{2,}$)', modifier), 'i');
```